joomla/joomla-cms
# [5.0] Add basic support for conditional dark mode themes in our editors

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/42148 and https://github.com/joomla/joomla-cms/issues/41801 .

### Summary of Changes
Supports dark mode detection for both tinymce and codemirror. Unlike the template which will actually flick over immediately once the OS is changed this will not change until the page is next reloaded. For codemirror this may be possible in the future as outlined in the todo in the code comment. But at the moment I don't believe this is possible for tinymce.

I've reused the same data attribute that @Fedik is suggesting in https://github.com/joomla/joomla-cms/pull/42221 (but I still disagree with the concept of a toggle in that PR - but this doesn't block that or alter that PR in any way). But the data attribute will be required on the body element to support this

### Testing Instructions
Open the editor in both light and dark modes. Before it would always open in light mode. Now it will match whatever the template is using on page load.

Note that casseopia which doesn't support automatic dark mode remains having editors only loaded in light mode.


### Link to documentations
Please select:

- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/206
- [] No documentation changes for manual.joomla.org needed
